salalib/shapemap.cpp

```diff
@@ -1674,7 +1677,9 @@ std::vector<int> ShapeMap::polyInPolyList(int polyref, double tolerance) const
                                     Line lineb = Line(polyb.m_points[shaperefb.m_polyrefs[kk]],polyb.m_points[((shaperefb.m_polyrefs[kk]+1)%polyb.m_points.size())]);
                                     if (intersect_region(line,lineb)) {
                                        if (intersect_line(line,lineb,tolerance)) {
-                                          if (testedlist.add(shaperefb.m_shape_ref) != -1) {
+                                          auto iterInternal = std::lower_bound(testedlist.begin(), testedlist.end(), shaperefb.m_shape_ref);

```

Comment:
Is `lower_bound` the correct algo here? That will also values that are bigger than the target value, not just exact matches.

Reply:
Yes, because I also added a check for that underneath, but this makes it exactly the same as the findBinary function we already have so I'll switch to that